test/jdk/javax/swing/JPopupMenu/MouseDragPopupTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JPopupMenu;
+
+/*
+ * @test
+ * @bug 8315655
+ * @requires (os.family == "mac")
+ * @key headful
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual MouseDragPopupTest
+ */
+public class MouseDragPopupTest {
+    static JFrame frame;
+    static JPanel panel;
+    static JPanel innerPanel;
+    static JPopupMenu menu;
+
+    public static void main(String[] args) throws Exception {
+        String instructions = """
+                1) Press and hold the right mouse button down on the JLabel with the text
+                "Right click and drag from here"
+
+                2) Move the mouse to the JLabel with the text "to here"
+
+                3) Observe that the popup menu assigned to the inner JPanel appears
+                (macOS) or does not appear (Windows, Linux)
+                """;
```

Comment:
The instruction is not clear as to what is expected to pass or fail the test. rephrase?